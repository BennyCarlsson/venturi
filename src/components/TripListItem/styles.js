import Heading from 'components/Heading'
import styled from 'styled-components'

export const TripListItemWrapper = styled.button`
  padding: 0;
  cursor: pointer;
  background-color: #ffffff;
  border: 1px solid #f3f3f3;
  border-radius: 8px;
  margin-bottom: 32px;
`

export const Top = styled.div`
  display: flex;
  align-items: center;
  height: 32px;
  background-color: #719fb0;
  border-radius: 8px;
`
export const StyledHeading = styled(Heading)`
  padding-left: 24px;
`

export const Content = styled.div`
  display: flex;
  padding: 8px 16px 24px 16px;
`

export const NameWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-left: 16px;
  padding-top: 3px;
  padding-bottom: 3px;
`
