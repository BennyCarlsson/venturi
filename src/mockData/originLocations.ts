export const locations = {
  locations: {
    loadingOrigin: false,
    loadingDestination: false,
    destinationLocations: [
      {
        id: '9021014014010000',
        idx: '1',
        name: 'Hålta kyrka, Kungälv',
        lat: '57.895206',
        lon: '11.827647',
        type: 0,
      },
      {
        id: '9021014012743000',
        idx: '2',
        name: 'Gårda, Mölndal',
        lat: '57.565347',
        lon: '12.092784',
        type: 0,
      },
      {
        id: '9021014002515000',
        idx: '3',
        name: 'Toftaåsgatan, Göteborg',
        lat: '57.645953',
        lon: '11.917170',
        type: 0,
      },
      {
        id: '9021014015934000',
        idx: '4',
        name: 'Hagaån, Orust',
        lat: '58.139155',
        lon: '11.625866',
        type: 0,
      },
      {
        id: '9021014021670000',
        idx: '5',
        name: 'Råane bro, Uddevalla',
        lat: '58.293814',
        lon: '11.984122',
        type: 0,
      },
      {
        id: '9021014021673000',
        idx: '6',
        name: 'Råane skola, Uddevalla',
        lat: '58.288996',
        lon: '11.988985',
        type: 0,
      },
      {
        id: '9021014021672000',
        idx: '7',
        name: 'Råane södra, Uddevalla',
        lat: '58.291360',
        lon: '11.986549',
        type: 0,
      },
      {
        id: 'A U Bååths väg, 439 94 Onsala',
        idx: '8',
        name: 'A U Bååths väg, 439 94 Onsala',
        lat: '57.390569',
        lon: '12.022587',
        type: 1,
      },
      {
        id: 'Allstakan Flaåsen, 671 98 Gunnarskog',
        idx: '9',
        name: 'Allstakan Flaåsen, 671 98 Gunnarskog',
        lat: '59.908834',
        lon: '12.510476',
        type: 1,
      },
      {
        id: 'Almvägen, 432 79 Tvååker',
        idx: '10',
        name: 'Almvägen, 432 79 Tvååker',
        lat: '57.044575',
        lon: '12.408116',
        type: 1,
      },
      {
        id: 'Alvägen, 432 79 Tvååker',
        idx: '11',
        name: 'Alvägen, 432 79 Tvååker',
        lat: '57.042903',
        lon: '12.404539',
        type: 1,
      },
      {
        id: 'Amandas väg, 432 77 Tvååker',
        idx: '12',
        name: 'Amandas väg, 432 77 Tvååker',
        lat: '57.060683',
        lon: '12.529336',
        type: 1,
      },
      {
        id: 'Anders Kristians väg, 432 77 Tvååker',
        idx: '13',
        name: 'Anders Kristians väg, 432 77 Tvååker',
        lat: '57.073115',
        lon: '12.368690',
        type: 1,
      },
      {
        id: 'Anemonvägen, 432 76 Tvååker',
        idx: '14',
        name: 'Anemonvägen, 432 76 Tvååker',
        lat: '57.012168',
        lon: '12.352572',
        type: 1,
      },
      {
        id: 'Angryd, 432 77 Tvååker',
        idx: '15',
        name: 'Angryd, 432 77 Tvååker',
        lat: '57.089359',
        lon: '12.540393',
        type: 1,
      },
      {
        id: 'Annelundsvägen, 432 76 Tvååker',
        idx: '16',
        name: 'Annelundsvägen, 432 76 Tvååker',
        lat: '57.042570',
        lon: '12.315941',
        type: 1,
      },
      {
        id: 'Askvägen, 432 79 Tvååker',
        idx: '17',
        name: 'Askvägen, 432 79 Tvååker',
        lat: '57.043370',
        lon: '12.403046',
        type: 1,
      },
      {
        id: 'Aspvägen, 432 79 Tvååker',
        idx: '18',
        name: 'Aspvägen, 432 79 Tvååker',
        lat: '57.044215',
        lon: '12.406894',
        type: 1,
      },
      {
        id: 'Augusts väg, 432 77 Tvååker',
        idx: '19',
        name: 'Augusts väg, 432 77 Tvååker',
        lat: '57.058598',
        lon: '12.526792',
        type: 1,
      },
      {
        id: 'Avavägen, 432 78 Tvååker',
        idx: '20',
        name: 'Avavägen, 432 78 Tvååker',
        lat: '57.047721',
        lon: '12.397797',
        type: 1,
      },
      {
        id: 'Rör, 432 77 Tvååker',
        idx: '21',
        name: 'Rör, 432 77 Tvååker',
        lat: '57.059937',
        lon: '12.445323',
        type: 1,
      },
      {
        id: 'Sik, 432 76 Tvååker',
        idx: '22',
        name: 'Sik, 432 76 Tvååker',
        lat: '57.001220',
        lon: '12.367845',
        type: 1,
      },
      {
        id: 'Påarps gård, Håcksvik',
        idx: '23',
        name: 'Påarps gård, Håcksvik',
        lat: '57.315473',
        lon: '13.147488',
        type: 1,
      },
      {
        id: 'Ängarna, 432 76 Tvååker',
        idx: '24',
        name: 'Ängarna, 432 76 Tvååker',
        lat: '57.028376',
        lon: '12.375692',
        type: 1,
      },
      {
        id: 'Ängvägen, 432 78 Tvååker',
        idx: '25',
        name: 'Ängvägen, 432 78 Tvååker',
        lat: '57.047217',
        lon: '12.402363',
        type: 1,
      },
      {
        id: 'Ästad, 432 77 Tvååker',
        idx: '26',
        name: 'Ästad, 432 77 Tvååker',
        lat: '57.066535',
        lon: '12.562155',
        type: 1,
      },
      {
        id: 'Ästad bondgård, 432 77 Tvååker',
        idx: '27',
        name: 'Ästad bondgård, 432 77 Tvååker',
        lat: '57.063245',
        lon: '12.555953',
        type: 1,
      },
      {
        id: 'Bäckvägen, 432 79 Tvååker',
        idx: '28',
        name: 'Bäckvägen, 432 79 Tvååker',
        lat: '57.038741',
        lon: '12.404889',
        type: 1,
      },
      {
        id: 'Ballstorp Kullaås, 535 93 Kvänum',
        idx: '29',
        name: 'Ballstorp Kullaås, 535 93 Kvänum',
        lat: '58.191256',
        lon: '13.253812',
        type: 1,
      },
      {
        id: 'Barrvägen, 432 79 Tvååker',
        idx: '30',
        name: 'Barrvägen, 432 79 Tvååker',
        lat: '57.035675',
        lon: '12.405806',
        type: 1,
      },
      {
        id: 'Beckasinvägen, 432 76 Tvååker',
        idx: '31',
        name: 'Beckasinvägen, 432 76 Tvååker',
        lat: '57.007296',
        lon: '12.358487',
        type: 1,
      },
      {
        id: 'Bergsjö Lövbodaåsarna, 681 94 Kristinehamn',
        idx: '32',
        name: 'Bergsjö Lövbodaåsarna, 681 94 Kristinehamn',
        lat: '59.421358',
        lon: '14.248622',
        type: 1,
      },
      {
        id: 'Bjällemad, 432 79 Tvååker',
        idx: '33',
        name: 'Bjällemad, 432 79 Tvååker',
        lat: '57.037329',
        lon: '12.411271',
        type: 1,
      },
      {
        id: 'Bjällemadvägen, 432 79 Tvååker',
        idx: '34',
        name: 'Bjällemadvägen, 432 79 Tvååker',
        lat: '57.037662',
        lon: '12.406678',
        type: 1,
      },
      {
        id: 'Bjären, 432 76 Tvååker',
        idx: '35',
        name: 'Bjären, 432 76 Tvååker',
        lat: '57.056485',
        lon: '12.351331',
        type: 1,
      },
      {
        id: 'Björka, 432 77 Tvååker',
        idx: '36',
        name: 'Björka, 432 77 Tvååker',
        lat: '57.086932',
        lon: '12.592782',
        type: 1,
      },
      {
        id: 'Björkäng, 432 76 Tvååker',
        idx: '37',
        name: 'Björkäng, 432 76 Tvååker',
        lat: '57.008672',
        lon: '12.362631',
        type: 1,
      },
      {
        id: 'Björkängs camping, 432 76 Tvååker',
        idx: '38',
        name: 'Björkängs camping, 432 76 Tvååker',
        lat: '57.009606',
        lon: '12.350226',
        type: 1,
      },
      {
        id: 'Björkängs havsbad, 432 76 Tvååker',
        idx: '39',
        name: 'Björkängs havsbad, 432 76 Tvååker',
        lat: '57.009418',
        lon: '12.350792',
        type: 1,
      },
      {
        id: 'Björkängs kvarn, 432 76 Tvååker',
        idx: '40',
        name: 'Björkängs kvarn, 432 76 Tvååker',
        lat: '57.013822',
        lon: '12.357813',
        type: 1,
      },
      {
        id: 'Björkängsvägen, 432 78 Tvååker',
        idx: '41',
        name: 'Björkängsvägen, 432 78 Tvååker',
        lat: '57.039127',
        lon: '12.398336',
        type: 1,
      },
      {
        id: 'Björkedalsvägen, 432 77 Tvååker',
        idx: '42',
        name: 'Björkedalsvägen, 432 77 Tvååker',
        lat: '57.072153',
        lon: '12.369822',
        type: 1,
      },
      {
        id: 'Björktrastvägen, 432 76 Tvååker',
        idx: '43',
        name: 'Björktrastvägen, 432 76 Tvååker',
        lat: '57.010676',
        lon: '12.356177',
        type: 1,
      },
      {
        id: 'Bjurbäcken Braåsen, 688 91 Storfors',
        idx: '44',
        name: 'Bjurbäcken Braåsen, 688 91 Storfors',
        lat: '59.577204',
        lon: '14.211325',
        type: 1,
      },
      {
        id: 'Blidsbergs Påarp, 523 77 Blidsberg',
        idx: '45',
        name: 'Blidsbergs Påarp, 523 77 Blidsberg',
        lat: '57.941051',
        lon: '13.480934',
        type: 1,
      },
      {
        id: 'Blomvägen, 432 79 Tvååker',
        idx: '46',
        name: 'Blomvägen, 432 79 Tvååker',
        lat: '57.043586',
        lon: '12.401392',
        type: 1,
      },
      {
        id: 'Bocklerud Laåsen, 672 94 Årjäng',
        idx: '47',
        name: 'Bocklerud Laåsen, 672 94 Årjäng',
        lat: '59.393186',
        lon: '11.994550',
        type: 1,
      },
      {
        id: 'Bofinksvägen, 432 76 Tvååker',
        idx: '48',
        name: 'Bofinksvägen, 432 76 Tvååker',
        lat: '57.009660',
        lon: '12.353300',
        type: 1,
      },
      {
        id: 'Borrås, 432 76 Tvååker',
        idx: '49',
        name: 'Borrås, 432 76 Tvååker',
        lat: '57.014434',
        lon: '12.441340',
        type: 1,
      },
      {
        id: 'Bosgårdsvägen, 432 79 Tvååker',
        idx: '50',
        name: 'Bosgårdsvägen, 432 79 Tvååker',
        lat: '57.044970',
        lon: '12.399747',
        type: 1,
      },
    ],
    originLocations: [
      {
        id: '9021014014010000',
        idx: '1',
        name: 'Hålta kyrka, Kungälv',
        lat: '57.895206',
        lon: '11.827647',
        type: 0,
      },
      {
        id: '9021014012743000',
        idx: '2',
        name: 'Gårda, Mölndal',
        lat: '57.565347',
        lon: '12.092784',
        type: 0,
      },
      {
        id: '9021014002515000',
        idx: '3',
        name: 'Toftaåsgatan, Göteborg',
        lat: '57.645953',
        lon: '11.917170',
        type: 0,
      },
      {
        id: '9021014015934000',
        idx: '4',
        name: 'Hagaån, Orust',
        lat: '58.139155',
        lon: '11.625866',
        type: 0,
      },
      {
        id: '9021014021670000',
        idx: '5',
        name: 'Råane bro, Uddevalla',
        lat: '58.293814',
        lon: '11.984122',
        type: 0,
      },
      {
        id: '9021014021673000',
        idx: '6',
        name: 'Råane skola, Uddevalla',
        lat: '58.288996',
        lon: '11.988985',
        type: 0,
      },
      {
        id: '9021014021672000',
        idx: '7',
        name: 'Råane södra, Uddevalla',
        lat: '58.291360',
        lon: '11.986549',
        type: 0,
      },
      {
        id: 'A U Bååths väg, 439 94 Onsala',
        idx: '8',
        name: 'A U Bååths väg, 439 94 Onsala',
        lat: '57.390569',
        lon: '12.022587',
        type: 1,
      },
      {
        id: 'Allstakan Flaåsen, 671 98 Gunnarskog',
        idx: '9',
        name: 'Allstakan Flaåsen, 671 98 Gunnarskog',
        lat: '59.908834',
        lon: '12.510476',
        type: 1,
      },
      {
        id: 'Almvägen, 432 79 Tvååker',
        idx: '10',
        name: 'Almvägen, 432 79 Tvååker',
        lat: '57.044575',
        lon: '12.408116',
        type: 1,
      },
      {
        id: 'Alvägen, 432 79 Tvååker',
        idx: '11',
        name: 'Alvägen, 432 79 Tvååker',
        lat: '57.042903',
        lon: '12.404539',
        type: 1,
      },
      {
        id: 'Amandas väg, 432 77 Tvååker',
        idx: '12',
        name: 'Amandas väg, 432 77 Tvååker',
        lat: '57.060683',
        lon: '12.529336',
        type: 1,
      },
      {
        id: 'Anders Kristians väg, 432 77 Tvååker',
        idx: '13',
        name: 'Anders Kristians väg, 432 77 Tvååker',
        lat: '57.073115',
        lon: '12.368690',
        type: 1,
      },
      {
        id: 'Anemonvägen, 432 76 Tvååker',
        idx: '14',
        name: 'Anemonvägen, 432 76 Tvååker',
        lat: '57.012168',
        lon: '12.352572',
        type: 1,
      },
      {
        id: 'Angryd, 432 77 Tvååker',
        idx: '15',
        name: 'Angryd, 432 77 Tvååker',
        lat: '57.089359',
        lon: '12.540393',
        type: 1,
      },
      {
        id: 'Annelundsvägen, 432 76 Tvååker',
        idx: '16',
        name: 'Annelundsvägen, 432 76 Tvååker',
        lat: '57.042570',
        lon: '12.315941',
        type: 1,
      },
      {
        id: 'Askvägen, 432 79 Tvååker',
        idx: '17',
        name: 'Askvägen, 432 79 Tvååker',
        lat: '57.043370',
        lon: '12.403046',
        type: 1,
      },
      {
        id: 'Aspvägen, 432 79 Tvååker',
        idx: '18',
        name: 'Aspvägen, 432 79 Tvååker',
        lat: '57.044215',
        lon: '12.406894',
        type: 1,
      },
      {
        id: 'Augusts väg, 432 77 Tvååker',
        idx: '19',
        name: 'Augusts väg, 432 77 Tvååker',
        lat: '57.058598',
        lon: '12.526792',
        type: 1,
      },
      {
        id: 'Avavägen, 432 78 Tvååker',
        idx: '20',
        name: 'Avavägen, 432 78 Tvååker',
        lat: '57.047721',
        lon: '12.397797',
        type: 1,
      },
      {
        id: 'Rör, 432 77 Tvååker',
        idx: '21',
        name: 'Rör, 432 77 Tvååker',
        lat: '57.059937',
        lon: '12.445323',
        type: 1,
      },
      {
        id: 'Sik, 432 76 Tvååker',
        idx: '22',
        name: 'Sik, 432 76 Tvååker',
        lat: '57.001220',
        lon: '12.367845',
        type: 1,
      },
      {
        id: 'Påarps gård, Håcksvik',
        idx: '23',
        name: 'Påarps gård, Håcksvik',
        lat: '57.315473',
        lon: '13.147488',
        type: 1,
      },
      {
        id: 'Ängarna, 432 76 Tvååker',
        idx: '24',
        name: 'Ängarna, 432 76 Tvååker',
        lat: '57.028376',
        lon: '12.375692',
        type: 1,
      },
      {
        id: 'Ängvägen, 432 78 Tvååker',
        idx: '25',
        name: 'Ängvägen, 432 78 Tvååker',
        lat: '57.047217',
        lon: '12.402363',
        type: 1,
      },
      {
        id: 'Ästad, 432 77 Tvååker',
        idx: '26',
        name: 'Ästad, 432 77 Tvååker',
        lat: '57.066535',
        lon: '12.562155',
        type: 1,
      },
      {
        id: 'Ästad bondgård, 432 77 Tvååker',
        idx: '27',
        name: 'Ästad bondgård, 432 77 Tvååker',
        lat: '57.063245',
        lon: '12.555953',
        type: 1,
      },
      {
        id: 'Bäckvägen, 432 79 Tvååker',
        idx: '28',
        name: 'Bäckvägen, 432 79 Tvååker',
        lat: '57.038741',
        lon: '12.404889',
        type: 1,
      },
      {
        id: 'Ballstorp Kullaås, 535 93 Kvänum',
        idx: '29',
        name: 'Ballstorp Kullaås, 535 93 Kvänum',
        lat: '58.191256',
        lon: '13.253812',
        type: 1,
      },
      {
        id: 'Barrvägen, 432 79 Tvååker',
        idx: '30',
        name: 'Barrvägen, 432 79 Tvååker',
        lat: '57.035675',
        lon: '12.405806',
        type: 1,
      },
      {
        id: 'Beckasinvägen, 432 76 Tvååker',
        idx: '31',
        name: 'Beckasinvägen, 432 76 Tvååker',
        lat: '57.007296',
        lon: '12.358487',
        type: 1,
      },
      {
        id: 'Bergsjö Lövbodaåsarna, 681 94 Kristinehamn',
        idx: '32',
        name: 'Bergsjö Lövbodaåsarna, 681 94 Kristinehamn',
        lat: '59.421358',
        lon: '14.248622',
        type: 1,
      },
      {
        id: 'Bjällemad, 432 79 Tvååker',
        idx: '33',
        name: 'Bjällemad, 432 79 Tvååker',
        lat: '57.037329',
        lon: '12.411271',
        type: 1,
      },
      {
        id: 'Bjällemadvägen, 432 79 Tvååker',
        idx: '34',
        name: 'Bjällemadvägen, 432 79 Tvååker',
        lat: '57.037662',
        lon: '12.406678',
        type: 1,
      },
      {
        id: 'Bjären, 432 76 Tvååker',
        idx: '35',
        name: 'Bjären, 432 76 Tvååker',
        lat: '57.056485',
        lon: '12.351331',
        type: 1,
      },
      {
        id: 'Björka, 432 77 Tvååker',
        idx: '36',
        name: 'Björka, 432 77 Tvååker',
        lat: '57.086932',
        lon: '12.592782',
        type: 1,
      },
      {
        id: 'Björkäng, 432 76 Tvååker',
        idx: '37',
        name: 'Björkäng, 432 76 Tvååker',
        lat: '57.008672',
        lon: '12.362631',
        type: 1,
      },
      {
        id: 'Björkängs camping, 432 76 Tvååker',
        idx: '38',
        name: 'Björkängs camping, 432 76 Tvååker',
        lat: '57.009606',
        lon: '12.350226',
        type: 1,
      },
      {
        id: 'Björkängs havsbad, 432 76 Tvååker',
        idx: '39',
        name: 'Björkängs havsbad, 432 76 Tvååker',
        lat: '57.009418',
        lon: '12.350792',
        type: 1,
      },
      {
        id: 'Björkängs kvarn, 432 76 Tvååker',
        idx: '40',
        name: 'Björkängs kvarn, 432 76 Tvååker',
        lat: '57.013822',
        lon: '12.357813',
        type: 1,
      },
      {
        id: 'Björkängsvägen, 432 78 Tvååker',
        idx: '41',
        name: 'Björkängsvägen, 432 78 Tvååker',
        lat: '57.039127',
        lon: '12.398336',
        type: 1,
      },
      {
        id: 'Björkedalsvägen, 432 77 Tvååker',
        idx: '42',
        name: 'Björkedalsvägen, 432 77 Tvååker',
        lat: '57.072153',
        lon: '12.369822',
        type: 1,
      },
      {
        id: 'Björktrastvägen, 432 76 Tvååker',
        idx: '43',
        name: 'Björktrastvägen, 432 76 Tvååker',
        lat: '57.010676',
        lon: '12.356177',
        type: 1,
      },
      {
        id: 'Bjurbäcken Braåsen, 688 91 Storfors',
        idx: '44',
        name: 'Bjurbäcken Braåsen, 688 91 Storfors',
        lat: '59.577204',
        lon: '14.211325',
        type: 1,
      },
      {
        id: 'Blidsbergs Påarp, 523 77 Blidsberg',
        idx: '45',
        name: 'Blidsbergs Påarp, 523 77 Blidsberg',
        lat: '57.941051',
        lon: '13.480934',
        type: 1,
      },
      {
        id: 'Blomvägen, 432 79 Tvååker',
        idx: '46',
        name: 'Blomvägen, 432 79 Tvååker',
        lat: '57.043586',
        lon: '12.401392',
        type: 1,
      },
      {
        id: 'Bocklerud Laåsen, 672 94 Årjäng',
        idx: '47',
        name: 'Bocklerud Laåsen, 672 94 Årjäng',
        lat: '59.393186',
        lon: '11.994550',
        type: 1,
      },
      {
        id: 'Bofinksvägen, 432 76 Tvååker',
        idx: '48',
        name: 'Bofinksvägen, 432 76 Tvååker',
        lat: '57.009660',
        lon: '12.353300',
        type: 1,
      },
      {
        id: 'Borrås, 432 76 Tvååker',
        idx: '49',
        name: 'Borrås, 432 76 Tvååker',
        lat: '57.014434',
        lon: '12.441340',
        type: 1,
      },
      {
        id: 'Bosgårdsvägen, 432 79 Tvååker',
        idx: '50',
        name: 'Bosgårdsvägen, 432 79 Tvååker',
        lat: '57.044970',
        lon: '12.399747',
        type: 1,
      },
    ],
  },
}
