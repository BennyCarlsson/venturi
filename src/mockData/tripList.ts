import { LocationType } from 'redux/locationsSlice'
import { TripListState } from 'redux/tripListSlice'

export const mockTripList: TripListState = {
  trips: [
    {
      name: 'TestName',
      destination: {
        id: '1',
        idx: '1',
        lat: '1',
        lon: '1',
        name: 'Destination Name',
        type: LocationType.CoordLocation,
      },
      origin: {
        id: '2',
        idx: '1',
        lat: '2',
        lon: '2',
        name: 'Origin name',
        type: LocationType.CoordLocation,
      },
    },
    {
      name: 'TestName2',
      destination: {
        id: '1',
        idx: '1',
        lat: '1',
        lon: '1',
        name: 'Destination Name',
        type: LocationType.CoordLocation,
      },
      origin: {
        id: '2',
        idx: '1',
        lat: '2',
        lon: '2',
        name: 'Origin name',
        type: LocationType.CoordLocation,
      },
    },
    {
      name: 'TestName3',
      destination: {
        id: '1',
        idx: '1',
        lat: '1',
        lon: '1',
        name: 'Destination Name',
        type: LocationType.CoordLocation,
      },
      origin: {
        id: '2',
        idx: '1',
        lat: '2',
        lon: '2',
        name: 'Origin name',
        type: LocationType.CoordLocation,
      },
    },
  ],
}
