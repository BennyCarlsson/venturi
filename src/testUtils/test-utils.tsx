import { configureStore } from '@reduxjs/toolkit';
import locationsReducer from '../redux/locationsSlice';
import testReducer from '../redux/testSlice';
import tripReducer from '../redux/tripSlice';
import dialogReducer from '../redux/dialogSlice';
import { render as rtlRender, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ReactElement } from 'react';
import { RootState } from 'redux/store';

type RenderOptionsType = { initialState?: Partial<RootState> } & Omit<RenderOptions, 'queries'>;
function render(ui: ReactElement, { initialState = {}, ...renderOptions }: RenderOptionsType) {
  const store = configureStore({
    reducer: {
      test: testReducer,
      trip: tripReducer,
      locations: locationsReducer,
      dialog: dialogReducer
    },
    preloadedState: initialState
  });
  function Wrapper({ children }: any) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
