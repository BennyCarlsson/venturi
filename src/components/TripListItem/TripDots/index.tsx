import { Dot, LetterCircle, TripDotsWrapper } from './styles'

const TripDots = () => {
  return (
    <TripDotsWrapper>
      <LetterCircle>A</LetterCircle>
      <Dot />
      <Dot />
      <Dot />
      <Dot />
      <LetterCircle>B</LetterCircle>
    </TripDotsWrapper>
  )
}

export default TripDots
