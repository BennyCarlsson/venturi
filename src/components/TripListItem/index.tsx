import Paragraph from 'components/Paragraph'
import { useAppDispatch } from 'hooks/redux'
import { Location } from 'redux/locationsSlice'
import { FontWeight } from 'types'
import { setCurrentTable } from 'redux/tripSlice'
import TripDots from './TripDots'
import {
  Content,
  NameWrapper,
  StyledHeading,
  Top,
  TripListItemWrapper,
} from './styles'

type TripListItemProps = {
  trip: { name: string; origin: Location; destination: Location }
}

const TripListItem = ({ trip, ...props }: TripListItemProps) => {
  const dispatch = useAppDispatch()

  return (
    <TripListItemWrapper
      onClick={() => {
        dispatch(
          setCurrentTable({
            origin: trip?.origin,
            destination: trip?.destination,
          })
        )
      }}
      {...props}
    >
      <Top>
        <StyledHeading
          color='#ffffff'
          fontSize={14}
          fontWeight={FontWeight.bold}
          headingType='h2'
        >
          {trip.name}
        </StyledHeading>
      </Top>
      <Content>
        <TripDots />
        <NameWrapper>
          <Paragraph
            color='#2f2e41'
            fontSize={14}
            fontWeight={FontWeight.medium}
          >
            {trip.origin?.name}
          </Paragraph>
          <Paragraph
            color='#2f2e41'
            fontSize={14}
            fontWeight={FontWeight.medium}
          >
            {trip.destination?.name}
          </Paragraph>
        </NameWrapper>
      </Content>
    </TripListItemWrapper>
  )
}

export default TripListItem
