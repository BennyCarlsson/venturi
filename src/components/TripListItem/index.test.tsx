import { mockTripList } from 'mockData/tripList'
import { render, screen } from 'testUtils/test-utils'
import TripListItem from '.'

describe('<TripListItem />', () => {
  const defaultProps: React.ComponentProps<typeof TripListItem> = {
    trip: mockTripList.trips[0],
  }
  beforeEach(() => {
    render(<TripListItem {...defaultProps} />, {})
  })

  it('should match snapshot', () => {
    const { asFragment } = render(<TripListItem {...defaultProps} />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    expect(screen).toBeTruthy()
  })

  it('should render trip name', () => {
    expect(
      screen.getByText(defaultProps.trip.name, { selector: 'h2' })
    ).toBeTruthy()
  })

  it('should render origin name', () => {
    expect(
      screen.getByText(defaultProps.trip.origin?.name as string, {
        selector: 'p',
      })
    ).toBeTruthy()
  })

  it('should render destination name', () => {
    expect(
      screen.getByText(defaultProps.trip.destination?.name as string, {
        selector: 'p',
      })
    ).toBeTruthy()
  })
})
