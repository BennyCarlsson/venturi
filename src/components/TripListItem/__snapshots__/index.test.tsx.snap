// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripListItem /> should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-jrAGrp kXgntB"
  >
    <div
      class="sc-kEjbxe ilXVtb"
    >
      <h2
        class="sc-jSgupP gKCGei sc-iqHYGH gbtLxu"
        color="#ffffff"
        font-size="14"
        font-weight="700"
      >
        TestName
      </h2>
    </div>
    <div
      class="sc-crrsfI cMVKCz"
    >
      <div
        class="sc-gsTCUz jEuoeG"
      >
        <div
          class="sc-hKgILt cRObAw"
        >
          A
        </div>
        <div
          class="sc-dlfnbm jpJEQU"
        />
        <div
          class="sc-dlfnbm jpJEQU"
        />
        <div
          class="sc-dlfnbm jpJEQU"
        />
        <div
          class="sc-dlfnbm jpJEQU"
        />
        <div
          class="sc-hKgILt cRObAw"
        >
          B
        </div>
      </div>
      <div
        class="sc-dQppl eIQGKt"
      >
        <p
          class="sc-bdfBwQ INFXx"
          color="#2f2e41"
          font-size="14"
          font-weight="500"
        >
          Origin name
        </p>
        <p
          class="sc-bdfBwQ INFXx"
          color="#2f2e41"
          font-size="14"
          font-weight="500"
        >
          Destination Name
        </p>
      </div>
    </div>
  </button>
</DocumentFragment>
`;
